include/profiler.h

```diff
@@ -142,20 +142,94 @@ void profiler_log_start(struct log_event_buf *buf);
 static inline void profiler_log_start(struct log_event_buf *buf) {}
 #endif
 
+/** @brief Encode and add uint32_t data type to a buffer.
+ *
+ * @warning The buffer must be initialized with @ref profiler_log_start
+ *          before calling this function.
+ *
+ * @param buf Pointer to the data buffer.
+ * @param data Data to add to the buffer.
+ */
+#ifdef CONFIG_PROFILER
+void profiler_log_encode_uint32(struct log_event_buf *buf, const uint32_t data);
```

Comment:
please don't use `const` for non pointer arguments

For pointer data `const` is a great thing when used in API. It informs that buffer should be threated as input. If there is no `const` it becomes output or input-output. No additional documentation is needed for direction. If argument names and function name are chose right documentation may not be needed altogether.

But for non-pointers this has no sense as these data are always input. Nothing can be returned this way. From API perspective using `const` to such arguments does not have any sense.